src/nwb_benchmarks/core/_streaming.py

```diff
@@ -57,21 +58,74 @@ def read_hdf5_fsspec_with_cache(
     return (file, byte_stream, tmpdir)
 
 
-def read_hdf5_nwbfile_fsspec_no_cache(
+def read_hdf5_fsspec_s3_no_cache(
+    s3_url: str,
+) -> Tuple[h5py.File, S3File]:
+    """Load the raw HDF5 file using fsspec with an S3 filesystem without a cache; does not load into pynwb."""
+    reset_lock()
+    fsspec.get_filesystem_class("s3").clear_instance_cache()
+    filesystem = fsspec.filesystem("s3", anon=True)
+    s3_form = s3_url.replace("https://dandiarchive.s3.amazonaws.com", "s3://dandiarchive")
```

Comment:
@rly in config the configuration, e.g., here

https://github.com/NeurodataWithoutBorders/nwb_benchmarks/blob/34505102e5899ffaf79cca89198c01cc8c9ce39c/src/nwb_benchmarks/benchmarks/params_remote_slicing.py#L3-L26

it looks like everything is configured with s3 URLs. I.e., am wondering whether the adaptations of the URIs is reversed. Calling `s3_url.replace("https://dandiarchive.s3.amazonaws.com", "s3://dandiarchive")` seems like to would be no-op since the input is already a S3 URL but then in the https function we don't seem to call the opposite `s3_url.replace("s3://dandiarchive", "https://dandiarchive.s3.amazonaws.com")`. I.e., I am wondering whether is currently using S3 URLs for both http and s3 tests

Reply:
I'll admit I always referred to the two forms as interchangeable - never occurred to me they might trigger different behavior/performance